fixtures/entrypoints-rpc-tests/tests/entrypoints.spec.ts

```diff
@@ -0,0 +1,825 @@
+		expect(output).toMatch(
+			'The `wrangler dev` session for service "bound" does export an entrypoint named "ThingEntrypoint"'
+		);
+	});
+});
+
+test("should support binding to wrangler session listening on HTTPS", async ({
+	dev,
+}) => {
+	// Start entry worker first, so the server starts with a stubbed service not
+	// found binding
+	const { url, session } = await dev({
+		"wrangler.toml": dedent`
+			name = "entry"
+			main = "index.ts"
+
+			[[services]]
+			binding = "SERVICE"
+			service = "bound"
+		`,
+		"index.ts": dedent`
+			export default {
+				async fetch(request, env, ctx) {
+					return env.SERVICE.fetch("http://placeholder/");
+				}
+			}
+		`,
+	});
+	let response = await fetch(url);
+	expect(await response.text()).toBe(
+		'[wrangler] Couldn\'t find `wrangler dev` session for service "bound" to proxy to'
+	);
+
+	// Start up the bound worker using HTTPS
+	const files: Record<string, string> = {
+		"wrangler.toml": dedent`
+			name = "bound"
+			main = "index.ts"
+		`,
+		"index.ts": dedent`
+			export default {
+				fetch() {
+					return new Response("secure");
+				}
+			};
+		`,
+	};
+	await dev(files, ["--local-protocol=https"]);
+
+	await waitFor(async () => {
+		const response = await fetch(url);
+		const text = await response.text();
+		expect(text).toBe("secure");
+	});
+});
+
+test("should throw if binding to version of wrangler without entrypoints support over HTTPS", async ({
+	dev,
+	isolatedDevRegistryPort,
+}) => {
+	// Start entry worker first, so the server starts with a stubbed service not
+	// found binding
+	const { url, session } = await dev({
+		"wrangler.toml": dedent`
+			name = "entry"
+			main = "index.ts"
+
+			[[services]]
+			binding = "SERVICE"
+			service = "bound"
+		`,
+		"index.ts": dedent`
+			export default {
+				async fetch(request, env, ctx) {
+					return env.SERVICE.fetch("http://placeholder/");
+				}
+			}
+		`,
+	});
+	let response = await fetch(url);
+	expect(await response.text()).toBe(
+		'[wrangler] Couldn\'t find `wrangler dev` session for service "bound" to proxy to'
+	);
+
+	// Simulate starting up the bound worker using HTTPS with an old version of Wrangler
+	response = await fetch(
+		`http://127.0.0.1:${isolatedDevRegistryPort}/workers/bound`,
+		{
+			method: "POST",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify({
+				protocol: "https",
+				mode: "local",
+				port: 0,
+				host: "localhost",
+				durableObjects: [],
+				durableObjectsHost: "localhost",
+				durableObjectsPort: 0,
+				// Intentionally omitting `entrypointAddresses`
+			}),
+		}
+	);
+	expect(response.status).toBe(200);
+	expect(await response.text()).toBe("null");
+
+	// Wait for error to be thrown
+	await waitFor(() => {
+		const output = session.getOutput();
+		expect(output).toMatch(
+			'Cannot proxy to `wrangler dev` session for service "bound" because it uses HTTPS. Please remove the `--local-protocol`/`dev.local_protocol` option.'
```

Comment:
If you _can_ bind to a HTTPS in new versions of Wrangler, should this error message be prompting you to upgrade rather than suggesting removing `--local-protocol=https`?